torch/distributed/pipelining/stage.py

```diff
@@ -1371,7 +1375,12 @@ def _shape_inference(
                 self.stage_index - 1,
             )
             dist.recv_object_list(
-                objects, src=self.prev_rank, group=self.group, device=self.device
+                objects,
+                src=_rank_to_global_rank(
+                    self.stage_index_to_group_rank[self.stage_index - 1], self.group
```

Comment:
you mentioned another PR to remove self.next_rank / prev_rank.  I think another option is to convert those to @property functions, and implement them with something like this code snippet.

```self.stage_index_to_group_rank[self.stage_index - 1]```

We should also audit whether we always return group-rank or global-rank from that helper and make sure that's consistent or theres a flag for controlling it if we need it.